src/components/profile.tsx

```diff
@@ -10,27 +16,68 @@ type Properties = {
 }
 
 const Profile: FC<Properties> = (properties) => {
+  const [anchorElement, setAnchorElement] = useState<HTMLElement>()
+  const open = Boolean(anchorElement)
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorElement(event.currentTarget)
+  }
+  const handleClose = () => {
+    // eslint-disable-next-line unicorn/no-useless-undefined
+    setAnchorElement(undefined)
+  }
+
   const navigate = useNavigate()
 
-  const onClick = () => {
+  const onLogout = () => {
     logout()
     navigate('/login')
   }
 
-  const Avatar = () => {
-    return (
-      <div className={styles['avatar']} aria-hidden="true">
-        <span>{properties.username.charAt(0).toUpperCase()}</span>
-      </div>
-    )
-  }
-
   return (
     <div className={styles['profile']}>
-      <Avatar />
-      <div className={styles['profile-actions']}>
-        <button onClick={onClick}>Logout</button>
-      </div>
+      <Box>
+        <IconButton onClick={handleClick} size="small">
+          <Avatar sx={{ bgcolor: lightGreen['A400'] }}>{properties.username.charAt(0).toUpperCase()}</Avatar>
+        </IconButton>
+      </Box>
+      <Menu
+        anchorEl={anchorElement}
+        id="profile-menu"
+        open={open}
+        onClose={handleClose}
+        onClick={handleClose}
+        PaperProps={{
+          elevation: 0,
+          sx: {
+            overflow: 'visible',
+            filter: 'drop-shadow(0px 2px 8px rgba(0,0,0,0.32))',
+            mt: 1.5,
+            // '^' arrow on top of the menu
+            '&:before': {
+              content: '""',
+              display: 'block',
+              position: 'absolute',
+              top: 0,
+              right: 14,
+              width: 10,
+              height: 10,
+              bgcolor: 'background.paper',
+              transform: 'translateY(-50%) rotate(45deg)',
+              zIndex: 0
+            }
+          }
+        }}
+        transformOrigin={{ horizontal: 'right', vertical: 'top' }}
+        anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
+      >
+        <MenuItem onClick={onLogout}>
+          <ListItemIcon>
+            <FontAwesomeIcon icon={faArrowRightFromBracket} size="sm" />
+          </ListItemIcon>
+          Logout
+        </MenuItem>
+      </Menu>
```

Comment:
This is one of the two places where we use MUI components instead of custom ones (because CSS is evil).